services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SwitchOperationsService.java

```diff
@@ -287,28 +287,61 @@ public SwitchPropertiesDto updateSwitchProperties(SwitchId switchId, SwitchPrope
             SwitchProperties switchProperties = switchPropertiesRepository.findBySwitchId(switchId)
                     .orElseThrow(() -> new SwitchPropertiesNotFoundException(switchId));
 
-            if (!update.isMultiTable() && update.isSwitchLldp()) {
-                throw new IllegalSwitchPropertiesException(
-                        String.format("Illegal switch properties combination for switch %s. 'switchLldp' property "
-                                + "can be set to 'true' only if 'multiTable' property is 'true'.", switchId));
-            }
+            validateSwitchProperties(switchId, update);
 
-            final boolean previousMultiTableFlag = switchProperties.isMultiTable();
-            final boolean previousLldpFlag = switchProperties.isSwitchLldp();
+            // must be called before updating of switchProperties object
+            final boolean isSwitchSyncNeeded = isSwitchSyncNeeded(switchProperties, update);
 
             switchProperties.setMultiTable(update.isMultiTable());
             switchProperties.setSwitchLldp(update.isSwitchLldp());
+            switchProperties.setSwitchArp(update.isSwitchArp());
             switchProperties.setSupportedTransitEncapsulation(update.getSupportedTransitEncapsulation());
 
             switchPropertiesRepository.createOrUpdate(switchProperties);
-            if (previousMultiTableFlag != update.isMultiTable() || previousLldpFlag != update.isSwitchLldp()) {
+            if (isSwitchSyncNeeded) {
                 carrier.requestSwitchSync(switchId);
             }
 
             return SwitchPropertiesMapper.INSTANCE.map(switchProperties);
         });
     }
 
+    private boolean isSwitchSyncNeeded(SwitchProperties current, SwitchProperties updated) {
+        return current.isMultiTable() != updated.isMultiTable()
+                || current.isSwitchLldp() != updated.isSwitchLldp()
+                || current.isSwitchArp() != updated.isSwitchArp();
+    }
+
+    private void validateSwitchProperties(SwitchId switchId, SwitchProperties updatedSwitchProperties) {
+        if (!updatedSwitchProperties.isMultiTable()) {
+            if (updatedSwitchProperties.isSwitchLldp()) {
+                throw new IllegalSwitchPropertiesException(
+                        String.format("Illegal switch properties combination for switch %s. 'switchLldp' property "
+                                + "can be set to 'true' only if 'multiTable' property is 'true'.", switchId));
+            }
+
+            if (updatedSwitchProperties.isSwitchArp()) {
```

Comment:
Can we combine this condition and the condition above?

Reply:
fixed